kolibri/plugins/management/assets/src/vuex/actions.js

```diff
@@ -6,7 +6,103 @@ function setSelectedClassroomId({ dispatch }, id) {
   dispatch('SET_SELECTED_CLASSROOM_ID', id);
 }
 
+function setSelectedGroupId({ dispatch }, id) {
+  dispatch('SET_SELECTED_GROUP_ID', id);
+}
+
+/*
+Used to wrap xhrs below, but note that it sets the onreadystatechange function,
+trampling whatever is there.
+ */
+function promiseWrapper(xhr) {
+  return new Promise((resolve, reject) => {
+    xhr.onreadystatechange = () => { // eslint-disable-line no-param-reassign
+      if (xhr.readyState === 4 && xhr.status === 200) {
+        resolve();
+      } else if (xhr.readyState === 4 && xhr.status !== 200) {
+        reject();
+      }
+    };
+  });
+}
+
+// An action for setting up the initial state of the app by fetching data from the server
```

Comment:
This is fine for now. However, eventually we should pre-load the initial state, similar to the list of nav-bar items. (Especially because it requires 3 requests.)